opensearch-operator/api/v1/opensearch_types.go

```diff
@@ -119,6 +117,15 @@ type TlsSecret struct {
 	Key        *string `json:"key,omitempty"`
 }
 
+type SecurityConfig struct {
+	// Secret that contains the differnt yml files of the opensearch-security config (config.yml, internal_users.yml, ...)
+	SecurityconfigSecret corev1.LocalObjectReference `json:"securityConfigSecret,omitempty"`
```

Comment:
Is the expectation here that the user builds these config files themselves with the appropriate usernames and passwords?  Would we consider creating the configs for the user (https://github.com/rancher-sandbox/opni-opensearch-operator/blob/main/pkg/resources/opensearch/config.go#L128-L163 is how I did it in the other operator for reference)?

Reply:
That is the expectation. In all projects where I used and use opensearch I need to configure custom users and roles so the approach you did would not be enough.
My idea was to extend it in a later version so that users/roles can be specified via CRDs (something like `kind: OpenSearchUser` or `kind: OpenSearchRole`) to make the management of it more kubernetes-native.

Reply:
Understood.  One approach might be to use the interal_users config just for initial population of users (admin and ops user), and then reconcile additional users directly against the the users API rather than going through the config.